### PR TITLE
feat: adding value_to_num mapping rules

### PR DESCRIPTION
**Pull request by Hardikl**
my prometheus url:
http://10.249.29.132:9090/graph?g0.expr=volume_new_status&g0.tab=1&g0.stacked=0&g0.range_input=1h

### CODE REVIEW

File: cmd/poller/plugin/README.md [hunk 1]
```diff
@@ -323,3 +324,35 @@ value_mapping: status state up,sleeping,unknown,down
...
+Note that you don't define the numeric values yourself, instead, you only provide the possible (expected) values, the plugin will map each value to its index in the rule.
+
+Rule syntax:
+
+```yaml
+value_to_num: METRIC LABEL ZAPI_VALUE REST_VALUE `N`
+# maps values of LABEL to 1 if it is ZAPI_VALUE or REST_VALUE
+# otherwise value of METRIC is set to N
+```
+The default value `N` is optional, if no default value is given and the label value does not match to any of the given values, the metric value will not be set.
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/poller/plugin/README.md:342]: done.

File: cmd/poller/plugin/README.md [hunk 2]
```diff
@@ -323,3 +324,35 @@ value_mapping: status state up,sleeping,unknown,down
 value_mapping: status state up `1`
 # metric value will be set to 0 if "state" is "up", otherwise to **1**
 ```
+
+
+## value_to_num
+
+Maps values of a given label to a numeric metric (of type `uint8`). This is sometimes handy to manipulate the data in the DB or Grafana (e.g. change color based on status or create alert).
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/poller/plugin/README.md:331]: value_mapping: status state online `1`
    value_to_num: status state online online `0`

if both the rules are define like this, value_to_num mapping would be applied, means when state value is online, status metric will show as 1.

**Comment by vgratian** [comment]
I got a bit confused seeing examples like: 
```yaml
value_to_num: status state online online `0`
```

does this mean that "ok value" of `status` is `online`, and for `state` - `online` as well?

**Comment by Hardikl** [comment]
This is the syntax: metric label zapi_value rest_value `default_value`
status state online online `0`

So, first online is possible zapi value in OK state and second online is possible REST value in OK state. To minimize the changes while REST implementation.

Will add comment in all yamls as well to better readability.

**Review by cgrinds — APPROVED** [review]

File: conf/zapi/cdot/9.8.0/volume.yaml [hunk 3]
```diff
@@ -57,6 +57,8 @@ counters:
 plugins:
   LabelAgent:
     value_mapping: status state online `1`
+    # metric label zapi_value rest_value `default_value`
+    value_to_num: new_status state online online `0`
```
Comment (rahulguptajss) [inline conf/zapi/cdot/9.8.0/volume.yaml:61]: does dashboard also needs to change or also include new_status value?

Comment (Hardikl) [reply conf/zapi/cdot/9.8.0/volume.yaml:61]: Yes, dashboard also need to be changed to consume new_status metric, it will be in next PR.

**Review by rahulguptajss — APPROVED** [review]

**Pull request merged**